caravel/assets/spec/javascripts/sqllab/reducers_spec.js

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import * as r from '../../../javascripts/SqlLab/reducers'
+import * as actions from '../../../javascripts/SqlLab/actions';
+import { describe, it } from 'mocha';
+import { expect } from 'chai';
+
+describe('sqlLabReducer', () => {
+  describe('CLONE_QUERY_TO_NEW_TAB', () => {
+    const testQuery = { sql: "SELECT * FROM...", dbId: 1, id: 1 };
+    const state = Object.assign(r.initialState, { queries: [ testQuery ] });
+    const newState = r.sqlLabReducer(state, actions.cloneQueryToNewTab(testQuery))
+
+    it('should have at most one more tab', () => {
+      expect(newState.queryEditors).have.length(2);
+      //expect(newState.queryEditors.filter((qe) => qe.sql == testQuery.SQL).length).to.equal(2)
+    });
+  
+    it('should have the same SQL as the cloned query', () => {
+      expect(newState.queryEditors[1].sql).to.equal(testQuery.sql);
+    });
+
+    it('should prefix the new tab title with "Copy of"', () => {
+      expect(newState.queryEditors[1].title).to.include("Copy of");
```

Comment:
Assuming that our new queryEditor will always be the last element is all well and good, but seems terribly hacky to me. Any suggestions or are you fine with it?